### ISSUE TITLE
[Bug]: Default register can throw unhandled error

### ISSUE THREAD

**Issue by maiconcarraro** [serwist/serwist#209]
### Provide environment information

```json
{
    "serwist": "^9.0.9",
    "@serwist/next": "^9.0.9",
}
```

### Which project is this issue for?

@serwist/window

### Link to reproduction - Issues with a link to complete (but minimal) reproduction code help us address them faster

-

### To reproduce

It can happen eventually, no specific steps.

### Describe the bug

On Sentry it's reporting some Rejected for the serviceWorker register:

![image](https://github.com/user-attachments/assets/26ba1783-4895-4ce5-9e03-04200316e0eb)

Related to this code: https://github.com/serwist/serwist/blob/6d1c9f08d37ac40c3bc7d26e4b820a4c36c16fc0/packages/window/src/Serwist.ts#L303-L323

### Expected behavior

I know it's possible to set `register: false` and implement myself the expected register handling any error, but opening this issue to understand/get context if it's intentional, or if possible to add another prop to automatically handle it. I can send a PR based in your interest and availability.

**Comment by DuCanhGH** [comment]
Ah, `@serwist/next`'s sw-entry.ts does not check whether it is in a secure context. So when your site is visited in incognito mode or not delivered via TLS, this code will always throw due to the fact that `navigator.serviceWorker` is undefined. I should fix this soon.

**Comment by roderickhsiao** [comment]
I know we are not actively maintaining https://github.com/DuCanhGH/next-pwa but it seems like we are having same issue

**Comment by DuCanhGH** [comment]
Wait, having checked the two packages, I noticed that both sw-entry.ts do actually check if `navigator.serviceWorker` is defined or not. Could there be another reason? @maiconcarraro @roderickhsiao Do you have a more specific error message?

**Comment by maiconcarraro** [comment]
@DuCanhGH no stacktrace, but I can share the device model, it's an android 7 and chrome v59... maybe some API missing?

![image](https://github.com/user-attachments/assets/880100b8-eccb-45a8-a5ef-46fcdb6d9111)

**Comment by DuCanhGH** [comment]
@maiconcarraro yeah, that's a bit unfortunate, then. Chrome 59 supports service workers, so I don't see how `navigator.serviceWorker.register` would actually throw. Could it be that there's something too modern in your service worker? Can you check if your compiled service worker actually supports those ancient browsers? There's this problem with Next.js where it doesn't compile modules from `node_modules` (?). See https://github.com/serwist/serwist/issues/182 for more :)
